File: src/main/java/kitchenpos/menus/tobe/application/MenuGroupService.java
package kitchenpos.menus.tobe.application;

import kitchenpos.menus.tobe.application.dto.MenuGroupCreationResponseDto;
import kitchenpos.menus.tobe.application.dto.MenuGroupsCreationResponseDto;
import kitchenpos.menus.tobe.domain.menugroup.MenuGroup;
import kitchenpos.menus.tobe.domain.menugroup.MenuGroupRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class MenuGroupService {
    private final MenuGroupRepository menuGroupRepository;

    public MenuGroupService(final MenuGroupRepository menuGroupRepository) {
        this.menuGroupRepository = menuGroupRepository;
    }

    @Transactional
    public MenuGroupCreationResponseDto create(String name) {
        final MenuGroup menuGroup = MenuGroup.create(name);
        menuGroupRepository.save(menuGroup);
        return MenuGroupCreationResponseDto.of(menuGroup);
    }

    @Transactional(readOnly = true)
    public MenuGroupsCreationResponseDto findAll() {
        return MenuGroupsCreationResponseDto.of(menuGroupRepository.findAll());
    }
}

File: src/main/java/kitchenpos/menus/tobe/application/MenuGroupService.kt
package kitchenpos.menus.tobe.application

import kitchenpos.menus.tobe.domain.MenuGroup
import kitchenpos.menus.tobe.domain.MenuGroupName
import kitchenpos.menus.tobe.domain.MenuGroupRepository
import kitchenpos.menus.tobe.dto.out.MenuGroupResponse
import kitchenpos.menus.tobe.dto.out.fromMenuGroup
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MenuGroupService(
    private val menuGroupRepository: MenuGroupRepository
) {

    @Transactional
    fun createMenuGroup(name: String): MenuGroupResponse {
        val menuGroup = MenuGroup(MenuGroupName(name))

        return fromMenuGroup(menuGroupRepository.save(menuGroup))
    }

}
